msteams-platform/concepts/bots/bots-files.md

```diff
@@ -0,0 +1,215 @@
+
+```csharp
+
+// This sample dialog shows two simple flows:
+// 1) A silly example of receiving a file from the user, processing the key elements,
+//    and then constructing the attachment and sending it back.
+// 2) Creating a new file consent card requesting user permission to upload a file.
+private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<object> result)
+{
+    var replyMessage = context.MakeMessage();
+    Attachment returnCard;
+
+    var message = await result as Activity;
+
+    // Check to see if the user is sending the bot a file.
+    if (message.Attachments != null && message.Attachments.Any())
+    {
+        var attachment = message.Attachments.First();
+
+        if (attachment.ContentType == FileDownloadInfo.ContentType)
+        {
+            FileDownloadInfo downloadInfo = (attachment.Content as JObject).ToObject<FileDownloadInfo>();
+            if (downloadInfo != null)
+            {
+                returnCard = CreateFileInfoAttachment(downloadInfo, attachment.Name, attachment.ContentUrl);
+                replyMessage.Attachments.Add(returnCard);
+            }
+        }
+    }
+    else
+    {
+        // Illustrates creating a file consent card.
+        returnCard = CreateFileConsentAttachment();
+        replyMessage.Attachments.Add(returnCard);
+    }
+    await context.PostAsync(replyMessage);
+}
+
+
+private static Attachment CreateFileInfoAttachment(FileDownloadInfo downloadInfo, string name, string contentUrl)
+{
+    FileInfoCard card = new FileInfoCard()
+    {
+        FileType = downloadInfo.FileType,
+        UniqueId = downloadInfo.UniqueId
+    };
+
+    Attachment att = card.ToAttachment();
+    att.ContentUrl = contentUrl;
+    att.Name = name;
+
+    return att;
+}
+
+private static Attachment CreateFileConsentAttachment()
+{
+    JObject acceptContext = new JObject();
+    // Fill in any additional context to be sent back when the user accepts the file.
+
+    JObject declineContext = new JObject();
+    // Fill in any additional context to be sent back when the user declines the file.
+
+    FileConsentCard card = new FileConsentCard()
+    {
+        AcceptContext = acceptContext,
+        DeclineContext = declineContext,
+        SizeInBytes = 102635,
+        Description = "File description"
+    };
+
+    Attachment att = card.ToAttachment();
+    att.Name = "Example file";
+
+    return att;
+}
+```
+
+### Notifying channel users
+We recommend using the Microsoft Graph APIs to perform read and write operations with a team's SharePoint folders. Read more [here](https://developer.microsoft.com/en-us/graph/docs/concepts/onedrive-concept-overview). You can also subscribe, via webhooks, to receive updates in real time whenever users upload files to its SharePoint. Read more about webhooks [here](https://developer.microsoft.com/en-us/graph/docs/api-reference/v1.0/resources/webhooks).
```

Comment:
As I said in my comments in Teams, this is not a realistic recommendation because you need Group.ReadWrite.All.

We need to support this in channels as a first-class scenario and not recommend this as the workaround.

Reply:
After discussion with Bill, I've removed this until we have a better story for channels.